lnwallet/channel.go

```diff
@@ -8441,8 +8663,9 @@ func (lc *LightningChannel) MaxFeeRate(maxAllocation float64) chainfee.SatPerKWe
 
 	// The maximum fee depends on the available balance that can be
 	// committed towards fees. It takes into account our local reserve
-	// balance.
-	availableBalance, weight := lc.availableBalance()
+	// balance. We do not account for a fee buffer here because that is
+	// exactly why it was introduced to react for sharp fee changes.
+	availableBalance, weight := lc.availableBalance(AdditionalHtlc)
```

Comment:
Hmm before laying out why I used `AdditionalHtlc` I try to explain why I don't used the other two in this case:

Why `FeeBuffer` was NOT used:

We are talking about updating the fee of the commitment here, that's one of the reasons the feebuffer was introduced which we keep when sending HTLCs? It's also referred as `fee spike buffer` in the Bolts. So when we would also use the buffer here we would not really be using it when feespikes happen ? Sure this means when we are drained locally and we update the fee, we might not be able to send further htlcs in the next commitment dance but because we use `AdditionalHtlc` we can still rebalance this HTLC and fill up local again
 
```
SHOULD NOT offer amount_msat if, after adding that HTLC to its commitment transaction, its remaining balance doesn't allow it to pay the commitment transaction fee when receiving or sending a future additional non-dust HTLC while maintaining its channel reserve. It is recommended that this "fee spike buffer" can handle twice the current feerate_per_kw to ensure predictability between implementations.
``` 

Why `NoBuffer` was NOT used:

Especially the edge case where we use 1 for `maxAllocation` and don't keep space for another HTLC, we might make the Channel unusable (only dust-htlcs might be sent) because no party can add an HTLC to the channel.

Thats why I decided to introduce this new Buffer type: `AdditionalHtlc `

But I might misunderstood something here.

Reply:
> So when we would also use the buffer here we would not really be using it when feespikes happen ? 

Exactly, `update_fee` is the place to use the fee buffer.

> Sure this means when we are drained locally and we update the fee, we might not be able to send further htlcs in the next commitment dance but because we use AdditionalHtlc we can still rebalance this HTLC and fill up local again

I'm not sure about this. Suppose we wanna rebalance, another round of `update_add_htlc` is sent and received. However, even this one htlc cannot be added because we require `FeeBuffer` when adding and receiving HTLCs?

Reply:
> I'm not sure about this. Suppose we wanna rebalance, another round of update_add_htlc is sent and received. However, even this one htlc cannot be added because we require FeeBuffer when adding and receiving HTLCs?

We do not enforce the buffer when receiving an htlc, because we are not in control as soon as the Peer adds an HTLC to our Commitment we can only accept it, otherwise the channel is borked.

> Exactly, update_fee is the place to use the fee buffer.

Do you think `AdditionalHTLC` is the right buffer type here then, I am still not sure if our understanding is the same 😅 ?